File: client-clojure/src/sirqul_io_t_platform/specs/achievement_short_response.clj
(ns sirqul-io-t-platform.specs.achievement-short-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def achievement-short-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :achievementId) int?
   (ds/opt :title) string?
   (ds/opt :description) string?
   (ds/opt :analyticsTag) string?
   (ds/opt :rankType) string?
   (ds/opt :rankIncrement) int?
   (ds/opt :validate) boolean?
   (ds/opt :appName) string?
   (ds/opt :active) boolean?
   (ds/opt :updated) int?
   (ds/opt :created) int?
   (ds/opt :tierCount) int?
   (ds/opt :minIncrement) int?
   (ds/opt :maxIncrement) int?
   (ds/opt :returning) string?
   })

(def achievement-short-response-spec
  (ds/spec
    {:name ::achievement-short-response
     :spec achievement-short-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/AchievementShortResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AchievementShortResponse._

case class AchievementShortResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
achievementId: Option[Long],
title: Option[String],
description: Option[String],
analyticsTag: Option[String],
rankType: Option[String],
rankIncrement: Option[Integer],
validate: Option[Boolean],
appName: Option[String],
active: Option[Boolean],
updated: Option[Long],
created: Option[Long],
tierCount: Option[Integer],
minIncrement: Option[Integer],
maxIncrement: Option[Integer],
returning: Option[String])

object AchievementShortResponse {
  import DateTimeCodecs._

  implicit val AchievementShortResponseCodecJson: CodecJson[AchievementShortResponse] = CodecJson.derive[AchievementShortResponse]
  implicit val AchievementShortResponseDecoder: EntityDecoder[AchievementShortResponse] = jsonOf[AchievementShortResponse]
  implicit val AchievementShortResponseEncoder: EntityEncoder[AchievementShortResponse] = jsonEncoderOf[AchievementShortResponse]
}
